src/php/lib/Grpc/Interceptor.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+namespace Grpc;
+
+
+class Interceptor{
+  public function UnaryUnary($method, $argument, $deserialize,
+                             array $metadata = [], array $options = [], $continuation){
+    return $continuation($method, $argument, $deserialize, $metadata, $options);
+  }
+
+  public function StreamUnary($method, $deserialize, array $metadata = [],
+                              array $options = [], $continuation){
+    return $continuation($method, $deserialize, $metadata, $options);
+  }
+
+  public function UnaryStream($method, $argument, $deserialize,
+                              array $metadata = [], array $options = [], $continuation){
+    return $continuation($method, $argument, $deserialize, $metadata, $options);
+  }
+
+  public function StreamStream($method, $deserialize,
+                               array $metadata = [], array $options = [], $continuation){
+    return $continuation($method, $deserialize, $metadata, $options);
+  }
+}
+
+class InterceptorChannel {
+  private $next = null;
+  private $interceptor;
+
+  public function __construct($channel, Interceptor $interceptor) {
+    $check_channel = $channel;
+    while($check_channel){
```

Comment:
I don't think this check is necessary. Just check if `$channel` itself is `null and if you want, just check if `$channel` is either `Grpc\Channel` or `Grpc\InterceptorChannel`. Checking for more things will just hinder opportunities to possibly mock the `Channel` object and feed it something else instead and increases the coupling of this piece of code with the implementation details of the rest of the system without meaningful advantages for the user.

Reply:
Done. removed this check.